src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -344,6 +381,31 @@ protected final void handleSelectedKey(final SelectionKey sk) throws IOException
 
   }
 
+  private void handleFullTasksCircularBuffer(Runnable r, int sleepMillis)
+  {
+    final Thread currentThread = Thread.currentThread();
+    if (eventThread == currentThread) {
+      int size = tasks.size();
+      if (size == tasks.capacity()) {
```

Comment:
Technically, it can be an `assert`, but as `tasks` is not expected to reach full capacity, introduced explicit check with throwing `IllegalStateException`.